services/supervisord/supervisord.go

```diff
@@ -406,6 +422,45 @@ func (s *Service) marshalConfig(tmpl *template.Template, settings *models.Settin
 	return b, nil
 }
 
+// addAlertManagerParam parses alertManagerURL
+// and extracts username and password from it to templateParams.
+func addAlertManagerParam(alertManagerURL string, templateParams map[string]interface{}) error {
+	templateParams["AlertmanagerURL"] = alertManagerURL
+	templateParams["AlertManagerUser"] = ""
+	templateParams["AlertManagerPassword"] = ""
+	if alertManagerURL == "" {
+		return nil
+	}
+	u, err := url.Parse(alertManagerURL)
+	if err != nil {
+		templateParams["AlertmanagerURL"] = ""
+
+		return errors.Wrap(err, "cannot parse AlertManagerURL")
+	}
+
+	if u.Opaque != "" || u.Host == "" {
+		templateParams["AlertmanagerURL"] = ""
+
+		return errors.Errorf("AlertmanagerURL parsed incorrectly as %#v", u)
+	}
+	if username := u.User.Username(); username != "" {
+		if password, ok := u.User.Password(); ok {
+			n := url.URL{
+				Scheme:   u.Scheme,
+				Host:     u.Host,
+				Path:     u.Path,
+				RawQuery: u.RawQuery,
+				Fragment: u.Fragment,
+			}
+			templateParams["AlertManagerUser"] = username
```

Comment:
is it possible to have AlertManager user without password?

Reply:
Nope, password is checked with `ok` value.

Reply:
No, I mean does vmalert support alert manager url's with username, but without password?

Reply:
Get it. Will fix.

Reply:
Changed logic.